aya/src/obj/btf/btf.rs

```diff
@@ -308,6 +308,30 @@ impl Btf {
         Ok(s.to_string_lossy())
     }
 
+    pub(crate) fn replace_string_at(&mut self, offset: u32, value: &str) -> Result<(), BtfError> {
+        let btf_header {
+            hdr_len,
+            mut str_off,
+            str_len,
+            ..
+        } = self.header;
+        str_off += hdr_len;
+
+        let c_str = CString::new(value).unwrap();
+        let new = c_str.as_bytes_with_nul();
+
+        if offset >= str_off + str_len || offset + new.len() as u32 > str_off + str_len {
+            return Err(BtfError::InvalidStringOffset {
+                offset: offset as usize,
+            });
+        }
+
+        let offset = offset as usize;
+        self.strings
+            .splice(offset..offset + new.len(), new.iter().cloned());
```

Comment:
What happens if new.len() is < old.len()? We leave random garbage bytes in there? I'm wondering if we shouldn't just add a new sanitized string and leave the old one as is, instead of having this replace_string_at at all.